tests/foreman/ui/test_discoveryrule.py

```diff
@@ -0,0 +1,168 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Foreman Discovery Rules"""
+from ddt import ddt
+from fauxfactory import gen_string
+from nailgun import entities
+from robottelo.common.decorators import data, run_only_on
+from robottelo.common.helpers import generate_strings_list
+from robottelo.test import UITestCase
+from robottelo.ui.factory import make_discoveryrule
+from robottelo.ui.locators import common_locators
+from robottelo.ui.session import Session
+
+
+@run_only_on('sat')
+@ddt
+class DiscoveryRules(UITestCase):
+    """Implements Foreman discovery Rules in UI."""
+    @classmethod
+    def setUpClass(cls):  # noqa
+        org_attrs = entities.Organization().create_json()
+        cls.hostgroup_name = entities.HostGroup().create_json()['name']
+        cls.org_name = org_attrs['name']
+        cls.org_id = org_attrs['id']
+        cls.org_label = org_attrs['label']
+
+        super(DiscoveryRules, cls).setUpClass()
+
+    @data(*generate_strings_list(len1=8))
+    def test_positive_create_discovery_rule_1(self, name):
+        """@Test: Create Discovery Rule
+
+        @Feature: Foreman Discovery
+
+        @Assert: Rule should be successfully created
+
+        """
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+
+    def test_positive_create_discovery_rule_2(self):
+        """@Test: Create Discovery Rule with 255 characters in name
+
+        @Feature: Foreman Discovery
+
+        @Assert: Rule should be successfully created
+
+        """
+        name = gen_string('alpha', 255)
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.search(name))
+
+    def test_negative_create_discovery_rule_1(self):
+        """@Test: Create Discovery Rule with 256 characters in name
+
+        @Feature: Foreman Discovery
+
+        @Assert: Error should be raised and rule should not be created
+
+        """
+        name = gen_string('alpha', 256)
+        with Session(self.browser) as session:
+            make_discoveryrule(session, name=name,
+                               hostgroup=self.hostgroup_name)
+            self.assertIsNotNone(self.discoveryrules.wait_until_element
+                                 (common_locators['name_haserror']))
```

Comment:
This is a stylistic thing, but can we avoid splitting apart function/method names from the parentheses indicating that the function/method is being called? Something like this would be good:

```python
            self.assertIsNotNone(self.discoveryrules.wait_until_element(
            	common_locators['name_haserror']
            ))
```

Or even this:

```python
            self.assertIsNotNone(self.discoveryrules.wait_until_element(
            	common_locators['name_haserror']))
```

Comment applies throughout this file.